course-mw/course-actors-common/src/main/java/org/sunbird/learner/actors/coursebatch/dao/impl/UserCoursesDaoImpl.java

```diff
@@ -327,4 +315,39 @@ public List<UserCourses> readV2(RequestContext requestContext, String userId, St
     return null;
   }
 
+  public long getCountOfActiveParticipants(RequestContext requestContext, String batchId) {
+    int ttl = Integer.parseInt(PropertiesCache.getInstance().getProperty(JsonKey.PARTICIPANTS_TTL));
+    int fetchSize = Integer.parseInt(PropertiesCache.getInstance().getProperty(JsonKey.PARTICIPANTS_FETCH_SIZE));
```

Comment:
move these lines after checking cachedCount is not empty... if redis has data these variables will not be used -- so no need to initialize before checking redis value...